nucypher/blockchain/eth/policies.py

```diff
@@ -112,54 +119,30 @@ def __find_ursulas(self, ether_addresses: List[str], target_quantity: int, timeo
 
             # Select an ether_address: Prefer the selection pool, then unknowns queue
             if ether_addresses:
-                ether_address = ether_addresses.pop()
+                ether_address = bytes(ether_addresses.pop(), encoding="ascii")
             else:
                 ether_address = unknown_addresses.popleft()
 
             try:
                 # Check if this is a known node.
-                selected_ursula = self.alice.known_nodes[ether_address]
+                selected_ursula = self.alice._known_nodes[ether_address]
 
             except KeyError:
                 # Unknown Node
-                self.alice.nodes_to_seek.add(ether_address)  # enter address in learning loop
+                self.alice.learn_about_specific_node(ether_address)  # enter address in learning loop
```

Comment:
Does this have a time delay?
The loop won't consume 100% CPU, right?

Reply:
Correct - it's called only every 10 seconds (a subsequent change will make this tuneable).  Also, it's mostly network I/O; it doesn't use much CPU.

Reply:
:heavy_check_mark: